src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -1372,8 +1412,7 @@ static int icap_download_boot_firmware(struct platform_device *pdev)
 	if (!ICAP_PRIVILEGED(icap))
 		return -EPERM;
 
-	/* Read dsabin from file system. */
-
+	/* Read xsabin first, if failed, try dsabin from file system. */
```

Comment:
why not just push all these xsabin first and dsabin second logic into find_firmware()?

Reply:
thanks, moved code into find_firmware